lib/tasks/omdb.rake

```diff
@@ -0,0 +1,250 @@
+      'tt0169547',  'tt0338013',  'tt0112573', 'tt2106476', 'tt0062622',
+      'tt0105236',  'tt0087843',  'tt0086190', 'tt0114709', 'tt0119698',
+      'tt0045152',  'tt0056172',  'tt0435761', 'tt0053604', 'tt0044741'
+    ]
+
+    ActiveRecord::Base.transaction do
+      top_100_movies_imdb_id.each do |imdb_id|
+        puts "Searching imdb id:#{imdb_id} using OMDB api..."
+        response = fetch_movie_details(imdb_id)
+
+        # Extracting movie details from the API response
+        actors = response['Actors'].split(', ')
+        awards = response['Awards']
+        box_office = response['BoxOffice']
+        countries = response['Country'].split(', ')
+        directors = response['Director'].split(', ')
+        dvd_release = response['DVD']
+        genres = response['Genre'].split(', ')
+        imdb_rating = response['imdbRating']
+        imdb_votes = response['imdbVotes']
+        imdb_id = response['imdbID']
+        languages = response['Language'].split(', ')
+        metascore = response['Metascore']
+        plot = response['Plot']
+        production = response['Production']
+        poster = response['Poster']
+        rated = response['Rated']
+        ratings = response['Ratings']
+        released = response['Released']
+        runtime = response['Runtime']
+        title = response['Title']
+        website = response['Website']
+        writers = response['Writer'].split(', ')
+        year = response['Year']
+
+        # Creating a new Movie object and saving it to the database
+        movie = Movie.new
+
+        movie.awards = awards
+        movie.box_office = box_office
+        movie.dvd_release = dvd_release
+        movie.imdb_rating = imdb_rating
+        movie.imdb_votes = imdb_votes
+        movie.id = imdb_id
+        movie.metascore = metascore
+        movie.plot = plot
+        movie.production = production
+        movie.poster = poster
+        movie.rated = rated
+        movie.released = released
+        movie.runtime = runtime
+        movie.title = title
+        movie.website = website
+        movie.year = year
+
+        # Calling methods to save related details to the database
+        save_actors(actors, imdb_id)
+        save_countries(countries, imdb_id)
+        save_directors(directors, imdb_id)
+        save_genres(genres, imdb_id)
+        save_languages(languages, imdb_id)
+        save_ratings(ratings, imdb_id)
+        save_writers(writers, imdb_id)
+
+        # Saving the Movie object
+        movie.save
+
+        puts "\n ************************************************ \n"
+      end
+    end
+  end
+
+  # Rake task to set up OMDB API base URL and key
+  desc 'OMDB base URL and API key setup for environment'
+  task omdb_api_setup: :environment do
+    # Task to set up OMDB API base URL and key in the environment
+
+    api_key = "1f724ce8"
+    base_url = "http://www.omdbapi.com/"
+
+    ENV['api_url'] = "#{base_url}?apikey=#{api_key}"
```

Comment:
move to the respective env file, right now you are using it in your local system then you can use `config/environments/development.rb`